### PR TITLE
fix(testbed): add defensive error handling for user-provided testbed loading

### LINKED ISSUE #480: feat(testbed): Add defensive error handling and negative tests for user-provided testbed loading
## Summary

Add defensive error handling and negative test cases for the user-provided testbed YAML loading feature introduced in PR #458.

## Context

PR #458 added support for users to provide a custom PyATS testbed.yaml. The implementation works correctly for valid inputs, but could benefit from more defensive programming to handle edge cases gracefully.

### Error Handling
- Add explicit `encoding="utf-8"` to file open operations
- Wrap `yaml.safe_load()` in try/except to catch `yaml.YAMLError` with user-friendly messages
- Validate that loaded YAML is a dict before attempting to access keys
- Handle edge case where YAML file is empty (returns `None`)

### Negative Test Cases
Add test coverage for:
- Malformed YAML syntax
- Empty YAML file
- YAML file containing wrong type (list instead of dict)
- YAML file with unexpected structure (e.g., `devices` key is a string instead of dict)

## Related

- PR #458: Initial implementation of user-provided testbed support

### PR DESCRIPTION
**Pull request by oboehmer**
## Summary

Add defensive error handling and negative test cases for the user-provided testbed YAML loading feature introduced in PR #458.

Fixes #480

### Error Handling Improvements in `_load_user_testbed()`

- Add explicit `encoding="utf-8"` to file open operations
- Wrap `yaml.safe_load()` in try/except to catch `yaml.YAMLError` with user-friendly error messages
- Validate that loaded YAML is a `dict` before attempting to access keys
- Handle edge case where YAML file is empty (returns `None`)
- Validate that `devices` key (if present) is a `dict`
- Guarantee `testbed` and `devices` keys exist as dicts to avoid code duplication in callers

### Negative Test Cases Added

| Test | Description |
|------|-------------|
| `test_malformed_yaml_syntax` | Malformed YAML (consolidated method) |
| `test_malformed_yaml_syntax_single_device` | Malformed YAML (single device method) |
| `test_empty_yaml_file` | Empty file (consolidated) |
| `test_empty_yaml_file_single_device` | Empty file (single device) |
| `test_yaml_with_only_comments` | File with only comments |
| `test_yaml_contains_list_instead_of_dict` | List at root level |
| `test_yaml_contains_string_instead_of_dict` | String at root level |
| `test_yaml_contains_integer_instead_of_dict` | Integer at root level |
| `test_devices_key_is_string_instead_of_dict` | `devices: "string"` |
| `test_devices_key_is_list_instead_of_dict` | `devices: [...]` |
| `test_devices_key_is_integer_instead_of_dict` | `devices: 123` |
| `test_error_message_includes_file_path` | Verifies helpful error messages |
| `test_load_user_testbed_guarantees_testbed_key` | Verifies `testbed` key created |
| `test_load_user_testbed_guarantees_devices_key` | Verifies `devices` key created |
| `test_load_user_testbed_guarantees_both_keys_minimal_yaml` | Verifies both keys from minimal YAML |

## Testing

All 43 testbed-related tests pass:
- 20 original tests in `test_testbed_generator.py`
- 8 existing tests in `test_testbed_generator_merge.py`
- 15 new tests in `test_testbed_generator_merge.py`

### CODE REVIEW

File: nac_test/pyats_core/execution/device/testbed_generator.py [hunk 1]
```diff
@@ -174,6 +164,59 @@ def generate_consolidated_testbed_yaml(
...
+            Validated testbed dictionary with 'testbed' and 'devices' keys
+            guaranteed to exist as dictionaries.
+
+        Raises:
+            ValueError: If the file contains invalid YAML, is empty,
+                or has unexpected structure.
+        """
+        try:
+            with open(base_testbed_path, encoding="utf-8") as f:
+                testbed = yaml.safe_load(f)
```
Comment (oboehmer) [inline nac_test/pyats_core/execution/device/testbed_generator.py]: to your comment on schema validation and my reply: We could fail early and just add this:

```suggestion
                from pyats.topology import loader
                testbed = yaml.safe_load(f)
                # also verify the schema early
                _ = loader.load(f)
```

**Review by aitestino — APPROVED** [review]
Hey @oboehmer, thanks for iterating on this. The changes look solid; approving as-is.

---

*P.S. — This comment was drafted using voice-to-text via Claude Code. If the tone comes across as overly direct or terse, please know that's just how it tends to phrase things. No offense or criticism is intended — this is purely an objective technical review of the PR. Thanks for understanding! 🙂*
